kong/globalpatches.lua

```diff
@@ -232,6 +233,15 @@ return function(options)
         return decode_args_real(str, max_args or MAX_DECODE_ARGS, ...)
       end
       -- ]
+
+      -- READ REQUEST BODY [
+      _G.ngx.req.read_body = function()
+        -- for the same request, only one `read_body` call is needed
+        if not ngx.ctx.body_read then
+          read_req_body()
```

Comment:
`ngx.req.read_body` only need to be called once in a request lifetime. Although multiple calls in the same request 
 don't re-read data to buffer multiple times, there are still some extra cost of check logic. So we could just avoid it to gain better performance.